crypto-ffi/src/metadata.rs

```diff
@@ -0,0 +1,113 @@
+    pub git_sha: String,
+    /// `true` when the source code differed from the commit at the most recent git hash
+    pub git_dirty: String,
+}
+
+/// Metadata describing the conditions of the build of this software.
+#[cfg(target_family = "wasm")]
+#[wasm_bindgen(inspectable)]
+pub struct BuildMetadata {
+    /// Build Timestamp
+    #[wasm_bindgen(readonly)]
+    pub timestamp: &'static str,
+    /// Whether this build was in Debug mode (true) or Release mode (false)
+    #[wasm_bindgen(readonly, js_name = "cargoDebug")]
+    pub cargo_debug: &'static str,
+    /// Features enabled for this build
+    #[wasm_bindgen(readonly, js_name = "cargoFeatures")]
+    pub cargo_features: &'static str,
+    /// Optimization level
+    #[wasm_bindgen(readonly, js_name = "optLevel")]
+    pub opt_level: &'static str,
+    /// Build target triple
+    #[wasm_bindgen(readonly, js_name = "targetTriple")]
+    pub target_triple: &'static str,
+    /// Git branch
+    #[wasm_bindgen(readonly, js_name = "gitBranch")]
+    pub git_branch: &'static str,
+    /// Output of `git describe`
+    #[wasm_bindgen(readonly, js_name = "gitDescribe")]
+    pub git_describe: &'static str,
+    /// Hash of current git commit
+    #[wasm_bindgen(readonly, js_name = "gitSha")]
+    pub git_sha: &'static str,
+    /// `true` when the source code differed from the commit at the most recent git hash
+    #[wasm_bindgen(readonly, js_name = "gitDirty")]
+    pub git_dirty: &'static str,
+}
+
+/// Returns build data for CoreCrypto
+#[cfg_attr(not(target_family = "wasm"), uniffi::export)]
+#[cfg_attr(target_family = "wasm", wasm_bindgen)]
+#[inline]
+pub fn build_metadata() -> BuildMetadata {
+    BuildMetadata {
+        #[cfg_attr(target_family = "wasm", expect(clippy::useless_conversion))]
+        timestamp: core_crypto::BUILD_METADATA.timestamp.into(),
+        #[cfg_attr(target_family = "wasm", expect(clippy::useless_conversion))]
+        cargo_debug: core_crypto::BUILD_METADATA.cargo_debug.into(),
+        #[cfg_attr(target_family = "wasm", expect(clippy::useless_conversion))]
+        cargo_features: core_crypto::BUILD_METADATA.cargo_features.into(),
+        #[cfg_attr(target_family = "wasm", expect(clippy::useless_conversion))]
+        opt_level: core_crypto::BUILD_METADATA.opt_level.into(),
+        #[cfg_attr(target_family = "wasm", expect(clippy::useless_conversion))]
+        target_triple: core_crypto::BUILD_METADATA.target_triple.into(),
+        #[cfg_attr(target_family = "wasm", expect(clippy::useless_conversion))]
+        git_branch: core_crypto::BUILD_METADATA.git_branch.into(),
+        #[cfg_attr(target_family = "wasm", expect(clippy::useless_conversion))]
+        git_describe: core_crypto::BUILD_METADATA.git_describe.into(),
+        #[cfg_attr(target_family = "wasm", expect(clippy::useless_conversion))]
+        git_sha: core_crypto::BUILD_METADATA.git_sha.into(),
+        #[cfg_attr(target_family = "wasm", expect(clippy::useless_conversion))]
```

Comment:
Instead of adding this attribute to every member, we can add it to the whole block, i.e.
```rust
    #[cfg_attr(target_family = "wasm", expect(clippy::useless_conversion))]
    BuildMetadata {
        timestamp: core_crypto::BUILD_METADATA.timestamp.into(),
        ...
```

Reply:
While it's possible to add it to the block, that unfortunately doesn't actually do what we want; clippy complains about an unfulfilled lint expectation in that case.